cmd/docs/docs.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package main
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/sylabs/singularity/cmd/internal/cli"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+func main() {
+	var dir string
+	var rootCmd = &cobra.Command{
+		ValidArgs: []string{"markdown", "man", "rest"},
+		Args:      cobra.ExactArgs(1),
+		Use:       "makeDocs {markdown | man | rest}",
+		Short:     "Generates Singularity documentation",
+		Run: func(cmd *cobra.Command, args []string) {
+			switch args[0] {
+			case "markdown":
+				sylog.Infof("Creating Singularity markdown docs at %s\n", dir)
+				if err := doc.GenMarkdownTree(cli.SingularityCmd, dir); err != nil {
+					sylog.Fatalf("Failed to create markdown docs for singularity\n")
+				}
+			case "man":
+				sylog.Infof("Creating Singularity man pages at %s\n", dir)
+				header := &doc.GenManHeader{
+					Title:   "singularity",
+					Section: "1",
+				}
+
+				// works recursively on all sub-commands (thanks bauerm97)
+				if err := doc.GenManTree(cli.SingularityCmd, header, dir); err != nil {
+					sylog.Fatalf("Failed to create man page for singularity\n")
```

Comment:
if this works on all man pages then this warning should be in plural: Failed to create **man pages** for Singularity